.github/workflows/pull-request.yaml

```diff
@@ -12,3 +12,22 @@ jobs:
     secrets: inherit
     with:
       charm-path: "charm"
+
+  # charm-pull-request runs unit tests on python 3.8
+  # Workaround to run unit-tests in python 3.10
+  unit-tests:
+    name: Unit tests (python 3.10)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```

Comment:
Shouldn't unit tests run with python 3.12 since it's a ubuntu@24.04 base?

Reply:
Fixed in subsequent commit.
I haven't pinned runs-on to `ubuntu-24.04` since `ubuntu-latest` will be soon rolled out to `ubuntu-24.04`
https://github.com/actions/runner-images/issues/10636

Reply:
Ok I pinned the runs-on to `ubuntu-24.04` and made change in tox command to actually run the unit tests